linked_list.rb

```diff
@@ -75,19 +99,25 @@ def find_nth_from_beginning(n)
   # list is sorted in ascending order
   def insert_ascending(value)
     current = @head
-    nextVal = nil
-    if current.next
-      nextVal = current.next.data
-      while nextVal <= value
-        current = current.next
+    prevVal = nil
+    # nextVal = nil
+      until value < current.data
+        puts "current.data is #{current.data}"
+        prevVal = current
+        current.next ? current = current.next : break
       end
-      temp = current.next
-      current.next = Node.new(value)
-      current.next.next = temp
-    else
-      current.next = Node.new(value)
+
+    new_node = Node.new(value)
+
+    if value > current.data && !current.next

```

Comment:
Check if current is nil first. :-)

Reply:
Scenarios to ensure your code works for: empty linked list, linked list has all values smaller than the value to insert, linked list with all values greater than the value to insert. In the empty linked list case and all values greater than value to insert case, head will need to get updated.